test/SILOptimizer/utf8_decoding_fastpath.swift

```diff
@@ -0,0 +1,79 @@
+// CHECK: ret
+public func decodeUBPAsUTF8(_ ptr: UnsafeBufferPointer<UInt8>) -> String {
+  return String(decoding: ptr, as: Unicode.UTF8.self)
+}
+
+// UnsafeMutableBufferPointer<UInt8>: we check that we call the merged function
+// =================================
+// CHECK-LABEL: define swiftcc {{.*}}s22utf8_decoding_fastpath16decodeUMBPAsUTF8ySSSrys5UInt8VGF{{.*}}
+// CHECK-NEXT: {{.*}} = tail call swiftcc {{.*}}s22utf8_decoding_fastpath15decodeUBPAsUTF8ySSSRys5UInt8VGFTm{{.*}}
+// CHECK: ret
+public func decodeUMBPAsUTF8(_ ptr: UnsafeMutableBufferPointer<UInt8>) -> String {
+  return String(decoding: ptr, as: Unicode.UTF8.self)
+}
+
+// this defined the merged function that handles both UBP and UMBP
+// CHECK-LABEL: define internal swiftcc {{.*}}s22utf8_decoding_fastpath15decodeUBPAsUTF8ySSSRys5UInt8VGFTm{{.*}}
+// CHECK-NOT: {{.*}} = call swiftcc {{.*}}_fromCodeUnits{{.*}}
+// CHECK: {{.*}} = call swiftcc {{.*}} @"$sSS18_fromUTF8Repairing{{.*}}
+// CHECK-NOT: {{.*}} = call swiftcc {{.*}}_fromCodeUnits{{.*}}
+// CHECK: ret
+
+// Array<UInt8>
+// ============
+// CHECK-LABEL: define swiftcc {{.*}}decodeArrayAsUTF8{{.*}}
+// CHECK-NOT: {{.*}} = call swiftcc {{.*}}_fromCodeUnits{{.*}}
+// CHECK: {{.*}} = call swiftcc {{.*}} @"$sSS18_fromUTF8Repairing{{.*}}
+// CHECK-NOT: {{.*}} = call swiftcc {{.*}}_fromCodeUnits{{.*}}
+// CHECK: ret
+public func decodeArrayAsUTF8(_ ptr: [UInt8]) -> String {
+  return String(decoding: ptr, as: Unicode.UTF8.self)
+}
+
+// UnsafeRawBufferPointer
+// ======================
+// CHECK-LABEL: define swiftcc {{.*}}decodeURBPAsUTF8{{.*}}
+// CHECK-NOT: {{.*}} = call swiftcc {{.*}}_fromCodeUnits{{.*}}
+// CHECK: {{.*}} = call swiftcc {{.*}} @"$sSS18_fromUTF8Repairing{{.*}}
+// CHECK-NOT: {{.*}} = call swiftcc {{.*}}_fromCodeUnits{{.*}}
+// CHECK: ret
+public func decodeURBPAsUTF8(_ ptr: UnsafeRawBufferPointer) -> String {
+  return String(decoding: ptr, as: Unicode.UTF8.self)
+}
+
+// UnsafeMutableRawBufferPointer
+// =============================
+// CHECK-LABEL: define swiftcc {{.*}}decodeUMRBPAsUTF8{{.*}}
+// CHECK-NOT: {{.*}} = call swiftcc {{.*}}_fromCodeUnits{{.*}}
+// CHECK: {{.*}} = call swiftcc {{.*}} @"$sSS18_fromUTF8Repairing{{.*}}
+// CHECK-NOT: {{.*}} = call swiftcc {{.*}}_fromCodeUnits{{.*}}
+// CHECK: ret
+public func decodeUMRBPAsUTF8(_ ptr: UnsafeMutableRawBufferPointer) -> String {
+  return String(decoding: ptr, as: Unicode.UTF8.self)
+}
+
+// Slice<UBP> doesn't actually work
+// ==========
+// CHECK-LABEL: define swiftcc {{.*}}decodeUBPSliceAsUTF8{{.*}}
+// CHECK-NOT: {{.*}} = call swiftcc {{.*}} @"$sSS18_fromUTF8Repairing{{.*}}
+// CHECK: {{.*}} = call swiftcc {{.*}}_fromCodeUnits{{.*}}
+// CHECK-NOT: {{.*}} = call swiftcc {{.*}} @"$sSS18_fromUTF8Repairing{{.*}}
+// CHECK: ret
+public func decodeUBPSliceAsUTF8(_ ptr: Slice<UnsafeBufferPointer<UInt8>>) -> String {
+  return String(decoding: ptr, as: Unicode.UTF8.self)
+}
```

Comment:
Could you file a bug and directly link it in the comment? Unless the test is XFAIL, I think it's better to disable the check than to test the undesired behavior. You can say something like:

```swift
// CHECK-LABEL: define swiftcc {{.*}}decodeUBPSliceAsUTF8{{.*}}
//
// ... comment linking and explaining bug, to be reenabled when fixed
// xCHECK-NOT: {{.*}} = call swiftcc {{.*}}_fromCodeUnits{{.*}}
// xCHECK: {{.*}} = call swiftcc {{.*}} @"$sSS18_fromUTF8Repairing{{.*}}
//
// CHECK: ret
```

Reply:
I'd say checking the undesired behavior is better because the test'll tell you if it changes for the better.

Reply:
@milseman up to you which way, I did file a bug

Reply:
I'll go with @jrose-apple's suggestion, but make it a little more explicit that you're checking against current behavior rather than desired behavior. Something like:

```
// FIXME(https://bugs.swift.org/browse/SR-9700):  _fromUTF8Repairing instead of _fromCodeUnits.
// Currently, Slice<UBP> doesn't actually work. Swap checked function below when resolved.
```

And then probably link to this specific test in the bug itself, so the resolver knows where to expect a change.